src/main/java/com/cs3365/rpgdapi/weapons/WeaponsController.java

```diff
@@ -64,4 +59,40 @@ public ResponseEntity get(@PathVariable("weaponID") UUID identifier)
                                          e.getMessage()));
         }
     }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<String> put(@RequestBody Weapon requestBody,
+                                      @PathVariable UUID identity) throws WeaponsException {
+        try {
+            requestBody.validate();
+        } catch(WeaponsException e) {
+            return ResponseEntity.badRequest().body(e.getMessage());
+        }
+
+        WeaponEntity weapon = weaponsService.findWeapon(identity);
+        if (weapon == null) {
+            return ResponseEntity.notFound().build();
+        }
+
+        weapon.setName(requestBody.getName());
+        weapon.setType(requestBody.getType());
+        weapon.setDescription(requestBody.getDescription());
+        weapon.setAttackPower(requestBody.getAttackPower());
+        weapon.setAttackType(requestBody.getAttackType());
+        weapon.setSpecialAbility(requestBody.getSpecialAbility());
+        weapon.setWeight(requestBody.getWeight());
+
+        try {
+            weaponsService.updateWeapon(weapon);
+        } catch(WeaponsException e) {
+            return ResponseEntity.badRequest()
+                    .body(String.format(
+                            "There was an issue updating the weapon in the database: %s",
+                            e.getMessage()));
+        }
+
+        return ResponseEntity.ok().body("");
```

Comment:
Here you would want to return the updated `WeaponEntity` object. The reason we would return the `WeaponEntity` object in the PUT is the same reason for the GET. This object is the only thing that will show the unique identifier of the weapon in the database.

Like I told Nick on his PR, if this were an enterprise application we would want to create something like a `WeaponsResponse` object, and transform `WeaponEntity` into `WeaponsResponse` before returning it to the end user. The reason for this being different objects manage the data at each layer of the stack.

For example, entity objects are objects used to interact with the database, models, such as the `Weapon` class are used in the business logic layer, and both request and response objects would need to be created for the presentation layer. Since this is a school project, it doesn't really matter. It's just important to note.

If we were using a NoSQL database like MongoDB, we would probably just use one object for every layer.